src/mbgl/shader/circle_vertex.cpp

```diff
@@ -0,0 +1,13 @@
+#include <mbgl/shader/circle_vertex.hpp>
+#include <mbgl/shader/shader.hpp>
+#include <mbgl/gl/gl.hpp>
+
+namespace mbgl {
+
+void CircleVertex::bind(const int8_t* offset) {
+    static_assert(sizeof(CircleVertex) == 4, "expected CircleVertex size");
+
+    MBGL_BIND_VERTEX_ATTRIBUTE(CircleVertex, a_pos, offset);
```

Comment:
[Sketched here](https://github.com/mapbox/mapbox-gl-native/commit/2c52f5959aca7f9bfa5f9ebad02534b3506c38d0) but currently stuck on a tricky problem with type erasing pointer-to-members.